### PR TITLE
More i18n and check_locales tool

### PR DESCRIPTION
**Pull request by alrusdi**

### CODE REVIEW

File: src/tools/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Terraforming-mars maintenance tools
+
+Tool directory is for specific scripts for project maintenance.
+
+Before use them build the project `npm run build`
```
Comment (kberg) [inline src/tools/README.md:5]: How are these getting built with npm run build, if npm run build doesn't include it? NB that I'm restructiring the tsconfig files in another PR.

File: src/tools/check_locales.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  warnings = [];
+  for (const localeName of localesToWarn) {
+    const trans: string = (translations as any)[localeName];
+    if ( ! trans && localeName === 'ru') {
+      warnings.push('\tmissing ' + localeName);
+    }
+  }
+  if (warnings.length > 0) {
+    console.log(sourceString);
+    for (const warning of warnings) {
```
Comment (kberg) [inline src/tools/check_locales.ts:42]: warnings.foreach((warning) => console.log(warning));

File: src/tools/check_locales.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+let sourceString: keyof typeof raw_translations;
+let warnings: Array<string>;
+
+for (sourceString in raw_translations) {
+  if ( ! raw_translations.hasOwnProperty(sourceString)) continue;
+  const translations = raw_translations[sourceString];
+  warnings = [];
+  for (const localeName of localesToWarn) {
+    const trans: string = (translations as any)[localeName];
+    if ( ! trans && localeName === 'ru') {
```
Comment (kberg) [inline src/tools/check_locales.ts:36]: Why is localeName === 'ru' important?

File: src/tools/check_locales.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+const LOCALES_DIR = path.resolve('./src/locales/');
+const locales: Array<string> = [];
+
+fs.readdirSync(LOCALES_DIR).forEach((localeName) => {
+  locales.push(localeName);
+});
+
+const args = process.argv.slice(2);
+let localesToWarn = locales;
+if (args) {
```
Comment (kberg) [inline src/tools/check_locales.ts:14]: You can simplify this:

```
if (args[0] === '--locales') {
  const localesToWarn = args[1].split(',');
  localesToWarn.forEach((locale) => {
    if (!locales.includes(locale)) {
      console.error(`Invalid locale ${locale}`);
      process.exit(1);
    }
  });
}
```

File: src/tools/README.md (hunk 1 above)
Comment (alrusdi) [reply src/tools/README.md:5]: I'll fix the doc when that PR will be merged

File: src/tools/check_locales.ts (hunk 2 above)
Comment (alrusdi) [reply src/tools/check_locales.ts]: Or even:
`console.warn(warnings.join('\n'));`

File: src/tools/check_locales.ts (hunk 4 above)
Comment (alrusdi) [reply src/tools/check_locales.ts]: Great addition!

File: src/tools/check_locales.ts (hunk 3 above)
Comment (alrusdi) [reply src/tools/check_locales.ts]: It is just mistake, thank you for noticing

**Review by chosta — APPROVED** [review]
This is great.

**Pull request merged**